server/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimUserEndpointsTests.java

```diff
@@ -797,6 +799,15 @@ public void testFindIdsByUserName() {
         assertEquals(joel.getId(), ((Map<String, Object>) results.getResources().iterator().next()).get("id"));
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testFindIdsByExternalId() {
+        SearchResults<?> results = endpoints.findUsers("id", "external_id eq \"b2f345ee-d893-44a9-b6ee-0abe865ff886\"", null, "ascending", 1, 100);
+        assertEquals(1, results.getTotalResults());
+        assertEquals(1, results.getSchemas().size());
+        assertEquals(joel.getId(), ((Map<String, Object>) results.getResources().iterator().next()).get("id"));
+    }
+
```

Comment:
Can you add a test for invalid external_id and an external id that doesn't exist?
```suggestion
    @SuppressWarnings("unchecked")
    @Test
    public void testFindIdsByExternalIdNonexistent() {
        SearchResults<?> results = endpoints.findUsers("id", "external_id eq \"does-not-exist\"", null, "ascending", 1, 100);
        assertEquals(0, results.getTotalResults());
        assertEquals(0, results.getSchemas().size());
    }
    
     @SuppressWarnings("unchecked")
    @Test
    public void testFindIdsByExternalIdWrongFormat() {
        SearchResults<?> results = endpoints.findUsers("id", "external_id eq \"#######\"", null, "ascending", 1, 100);
        assertEquals(0, results.getTotalResults());
        assertEquals(0, results.getSchemas().size());
    }
```